### ISSUE TITLE
Official Project Mentors for new Collaborators

### ISSUE THREAD

**Issue by MylesBorins** [nodejs/TSC#365]
I would like to suggest that we create an official mentorship program. To start I think it would be a good idea to give every individual being given a commit bit an official project mentor.

As a longer term goal I would love to see a broader mentor initiative where all members of the project, regardless of seniority, act as a mentor and mentee.

**Comment by mcollina** [comment]
I think this is a very positive initiative. Count me in!

**Comment by ofrobots** [comment]
+1. I'd be glad to participate as well.

**Comment by sam-github** [comment]
I've the time to do some mentorship.

**Comment by refack** [comment]
I'm also be happy to volunteer, in whatever position you deem fit...

**Comment by mhdawson** [comment]
Seems like a good initiative.

**Comment by jasnell** [comment]
Do you have a structure in mind for this? Mentorship programs work best when you have a good number of dedicated mentors who will stick with it. We have had issues with long term contributors staying consistently engaged enough. While I'm all for this, it would be good to have some specifics about how this would be administered and how we'll encourage mentors to follow through.

**Comment by dshaw** [comment]
At the Node.js Interactive 2017 Collaboration Summit, we discussed creating a Mentorship WG. I think chartering a Working Group to better define expectations and support both mentors and mentees would be an effective way to bootstrap this. I would expect that group to be chartered under the CommComm and from there support the mentorship needs across all of the Node.js project (TSC, CommComm, Board, Foundation staff).

I raised my hand to help bootstrap. Please join me if you want to dedicate several hours a month to support mentorship in Node.js.

Request for explicit feedback:
1) Should we form a Mentorship Working Group?
2) Should that WG be chartered under the CommComm?
3) Do you want to be a part of the Mentorship WB bootstrap effort?
4) Do you need a mentor?
5) Do you need mentee(s)?

**Comment by jasnell** [comment]
@dshaw... Count me in.

**Comment by jasnell** [comment]
To the specific questions:

Yes, we should have a mentorship wg
Having it under CommComm is fine but I'd like TSC buy in
Yes, No, and Yes.

**Comment by mcollina** [comment]
I do not think this needs to be chartered. A team should suffice. It is ok if this is under commcomm.
Yes, No and Yes as the last three questions.

**Comment by MylesBorins** [comment]
> Should we form a Mentorship Working Group?
I do not think it needs to be a chartered WG

> Should that WG be chartered under the CommComm?
I think it would make sense to ask the CommComm to come up with an initial proposal for how a mentorship program could be implemented, at that point we would need TSC buy in to follow through.

> Do you want to be a part of the Mentorship WB bootstrap effort?
yes

> Do you need a mentor?
I think everyone should have one, but do not need one during bootstrapping

> Do you need mentee(s)?
Yes please

**Comment by jasnell** [comment]
I would say that it makes more sense to ask the Foundation to find an outside expert in mentoring programs to come help us figure out the right path before deciding whether it should live in CommComm or TSC or whatnot.

**Comment by mhdawson** [comment]
Should we form a Mentorship Working Group?
- as others have said can just be a team
Should that WG be chartered under the CommComm?
- yes
Do you want to be a part of the Mentorship WB bootstrap effort?
- No, looks like there are already a number of people volunteering for the bootstrap and I trust them to move it foward
Do you need a mentor?
- I don't need one immediately but as James says its probably good if everybody has one.
Do you need mentee(s)
- yes

**Comment by dshaw** [comment]
Can someone point me to our definition of Team, Working Group and chartered Working Group? This is the second time, I have used the term "working group" and received pushback that the thing should be a "team".

**Comment by MylesBorins** [comment]
@dshaw you can find more info on the difference [in this doc](https://github.com/nodejs/TSC/blob/master/WORKING_GROUPS.md)

TLDR; chartering a working group is a lot of work, spinning up a team is fast

**Comment by sam-github** [comment]
@dshaw Other than the effort of spinning one up, there no longer appears to be much (if any) advantage to being a formal working group over a team. I think when the process was originally bootstrapped it was novel, up until then, there had been the core team and... everybody else. But bringing up WGs to focus on particular areas seemed a good idea, and some kind of ratification of the WGs existence and a stamp of approval by the core team seemed like an essential part of that process. Since those early days, non-WG teams have more often just been formed ad-hoc, they do good work, they get given github repo's under the nodejs org, meetings get put on the official calendar, people participate, and there doesn't seem to be any reason to get them ratified formally, its just more process.

If WGs get given votes on the TSC, or some other compelling benefit over just teams, the distinction may become more important again.

**Comment by mhdawson** [comment]
This is doc on WG's https://github.com/nodejs/TSC/blob/master/WORKING_GROUPS.md

The main difference is that WG's are formally delegated authority from the TSC.

**Comment by dshaw** [comment]
Thanks for the context. I'm going to continue on this tangent for a bit. If I should be creating a new issue instead for this discussion, please advise.

The concept that I see missing in [how we have defined working groups](https://github.com/nodejs/TSC/blob/master/WORKING_GROUPS.md) and what we are referring to colloquially as "teams" (not defined in the WG doc or in any of our other documents, as far as I can tell), is what we refer to as a subcommittee at the Board level. A subcommittee of the Board can either be short lived (like the Executive Director Selection Subcommittee) or long lived (like the Budget and Finance Subcommittee). A short-lived subcommittee hasn't required much in the way of a formal chartering, but it is specifically empowered by the body (the Board) to act independently on its behalf. There isn't the formal delegation of authority, but there is an empowerment of a small, focused team.

Maybe that's what folks are referring to as "team", but as an outsider, I'm having a hard time identifying the process I would go about to forming a group that has the TSC's blessing to work on special interests _on behalf of the TSC_ to even get to what might be worth considering a Working Group.

**Comment by jasnell** [comment]
@dshaw ... The confusion you're experiencing is certainly not unique or surprising. WGs and teams have long been poorly defined, and efforts to improve their definition have fallen flat. Even this idea that WGs have actual autonomy over their chartered work areas is not actually realized.

**Comment by MylesBorins** [comment]
Afaik simply opening an issue and expressing interest is enough to get a
team off the ground. From there things escalate as needed. The current
rules allow any member of the organization to make new repos, although care
should be taken when creating things that could be viewed as officially
endorsed.

All decisions by a team need to be approved by tsc

A chartered working group has been entrusted with ownership over a part of
the project and can act indpeendently of the tsc

On Oct 16, 2017 3:26 PM, "Dan Shaw" <notifications@github.com> wrote:

Thanks for the context. I'm going to continue on this tangent for a bit. If
I should be creating a new issue instead for this discussion, please advise.

The concept that I see missing in how we have defined working groups
<https://github.com/nodejs/TSC/blob/master/WORKING_GROUPS.md> and what we
are referring to colloquially as "teams" (not defined in the WG doc or in
any of our other documents, as far as I can tell), is what we refer to as a
subcommittee at the Board level. A subcommittee of the Board can either be
short lived (like the Executive Director Selection Subcommittee) or long
lived (like the Budget and Finance Subcommittee). A short-lived
subcommittee hasn't required much in the way of a formal chartering, but it
is specifically empowered by the body (the Board) to act independently on
its behalf. There isn't the formal delegation of authority, but there is an
empowerment of a small, focused team.

Maybe that's what folks are referring to as "team", but as an outsider, I'm
having a hard time identifying the process I would go about to forming a
group that has the TSC's blessing to work on special interests *on behalf
of the TSC* to even get to what might be worth considering a Working Group.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/nodejs/TSC/issues/365#issuecomment-337006797>, or mute
the thread

**Comment by Trott** [comment]
FWIW, the "team" terminology is GitHub's, which may be part of why it was never documented/defined. https://help.github.com/articles/about-teams/

**Comment by joyeecheung** [comment]
FWIW in my understanding, teams = people to ping, WG = people who have regular meetings

**Comment by jasnell** [comment]
A mentorship program has been initiated under the community committee. There does not appear to be anything further here to discuss. Can we close this?

**Comment by mhdawson** [comment]
I'd agree we should handle under the mentorship program and close this issue.  @MylesBorins you ok if we close ?

**Comment by fhinkel** [comment]
Closing. Feel free to reopen if needed.
